### PR TITLE
Treat a nonzero exit status as a failure

### LINKED ISSUE #3429: Test files that fail without output are not considered "failed"
See https://github.com/GothenburgBitFactory/taskwarrior/pull/3424

### PR DESCRIPTION
**Pull request by djmitche**
Fixes #3429. cc @cclauss!

@felixschurk what do you think? If this looks like the right fix, I can update the tests that now fail.

### CODE REVIEW

**Comment by djmitche** [comment]
For my reference later:
```
Failed:                        
eval.test                           1
lexer.test                          1
t.test                              1
tdb2.test                           1
util.test                           1
version.test                        1
view.test                           1
```

File: test/lexer.test.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@
 int main (int, char**)
 {
 #ifdef PRODUCT_TASKWARRIOR
-  UnitTest t (1253);
+  UnitTest t (1259);
```
Comment (ryneeverett) [inline test/lexer.test.cpp]: What is this about?

**Review by ryneeverett — APPROVED** [review]
Seems like the DIY test infrastructure has really been biting us lately and there are probably better solutions these days.

File: test/lexer.test.cpp (hunk 1 above)
Comment (djmitche) [reply test/lexer.test.cpp]: Oh, I think I need to update that again. It sets the number of tests expected -- part of TAP.

**Comment by djmitche** [comment]
> Seems like the DIY test infrastructure has really been biting us lately and there are probably better solutions these days.

Agreed, although thanks to @felixschurk and your work recently, it's a lot better than it has been. I'll add that to my list of things to talk about!

**Pull request merged**